packages/documentsearch-extension/src/index.ts

```diff
@@ -450,6 +450,12 @@ const extension: JupyterFrontEndPlugin<ISearchProviderRegistry> = {
       }
     });
 
+    app.shell.currentChanged?.connect(() => {
```

Comment:
Nice if it does

Reply:
Maybe this snippet could be updated to use `app.shell.currentChanged?` ?

https://github.com/jupyterlab/jupyterlab/blob/eb07560b542a1d3b6273f895bd897ba1de618e41/packages/documentsearch-extension/src/index.ts#L102-L111

Or do we need `activeChanged`? Or could we handle both `labShell.activeChanged` if available, otherwise default to `app.shell.currentChanged?` ?

Reply:
There is a slight difference: https://github.com/jupyterlab/lumino/blob/main/packages/widgets/src/focustracker.ts

For the case of notebook, `app.shell.currentChanged` will do as there is only one possible widget. But it may not be the case in other multidocument UI.

That said I don't have a strong opinion.